src/Entities/Entity.cpp

```diff
@@ -1325,6 +1325,12 @@ void cEntity::ScheduleMoveToWorld(cWorld * a_World, Vector3d a_NewPosition, bool
 
 bool cEntity::DetectPortal()
 {
+	// Don't let attached entities change worlds, like players riding a minecart
+	if ((m_AttachedTo != nullptr) || (m_Attachee != nullptr))
+	{
+		return false;
+	}
+
 	// If somebody scheduled a world change with ScheduleMoveToWorld, change worlds now.
```

Comment:
Does this mean calling `ScheduleMoveToWorld` won't work at all for players riding minecarts?

Reply:
I'm so glad we've got @peterbell10 reviewing the code; I tend to miss these "systematic" issues.

Reply:
ScheduleMoveToWorld behaves like before (teleports once you exit the vehicle), but that's not correct behavior either. I'll have to look closer at this one.

MoveToWorld teleports immediately, but messes up the client (can't break any blocks).

Reply:
Perhaps adding a call to `Detach` in `DoMoveToWorld` would fix that?
This condition could then be moved so it only covers the portal checking code and then it would be like vanilla, right?